src/components/ItemEditorPage/LeftPanel/Header/Header.tsx

```diff
@@ -16,10 +16,10 @@ export default class Header extends React.PureComponent<Props> {
   }
 
   handleBack = () => {
-    const { collection, isFromCollections, onNavigate } = this.props
+    const { collection, isFromCollections, isFromTPCollections, onNavigate } = this.props
     // If the user came from a collection details page, go back to the same page
-    if (collection && isFromCollections) {
-      onNavigate(locations.collectionDetail(collection.id))
+    if (collection && (isFromCollections || isFromTPCollections)) {
+      onNavigate(isFromCollections ? locations.collectionDetail(collection.id) : locations.thirdPartyCollectionDetail(collection.id))
```

Comment:
```suggestion
      let location = isFromCollections ? locations.collectionDetail : locations.thirdPartyCollectionDetail
      onNavigate(location(collection.id))
```
this might be a little bit cleaner, feel free to accept the suggestion, yours works perfect too.